plugins/animate/animate.cpp

```diff
@@ -18,19 +18,31 @@ bool animation_base::step()
     return false;
 }
 
+void animation_base::reverse()
+{}
+
+int animation_base::get_direction()
+{}
+
 animation_base::~animation_base()
 {}
 
-static constexpr const char *animate_custom_data_id = "animation-hook";
+static constexpr const char *animate_custom_data_fire     = "animation-hook-fire";
```

Comment:
Is there a reason why we need to differentiate the different types of animation? It seems like it should be enough to differentiate between custom data for map/unmap and for minimize/restore.

Reply:
> Is there a reason why we need to differentiate the different types of animation? It seems like it should be enough to differentiate between custom data for map/unmap and for minimize/restore.

Yes, the reason is to support multiple simultaneous animations. Consider the following:
An application maps a surface but unmaps it before the open animation completes. If there are different animations set for open and close, it can only know that 'a close animation is requested' and without knowing which one is requested, it doesn't know whether to reverse the current animation or start a new one. If the open and close animations are always the same, then it wouldn't really matter because it would always reverse.

Reply:
I see, thanks for the explanation.